### ISSUE TITLE
Macro not working with command including %u

### ISSUE THREAD

**Issue by tamis-laan** [newsboat/newsboat#917]
The following macro works when spawning a mpv youtube video with a max resolution of 720:
```
macro v set browser "mpv -ytdl-format='bestvideo[height<=?720]+bestaudio/best'" ; open-in-browser ; set  browser linkhandler
```
However when I use %u it stops working:
```
macro v set browser "mpv -ytdl-format='bestvideo[height<=?720]+bestaudio/best' %u" ; open-in-browser ; set  browser linkhandler
```
This is important because in the end I would like to detach the process as so:
```
macro v set browser "nohup mpv -ytdl-format='bestvideo[height<=?720]+bestaudio/best' %u &>/dev/null &" ; open-in-browser ; set  browser linkhandler
```
The strange thing is that the followoing does work:
```
macro v set browser "nohup mpv -ytdl-format=22 %u &>/dev/null &" ; open-in-browser ; set browser linkhandler 
```
Can anyone tell me what I'm doing wrong? Or is this some sort of bug?

**Comment by Minoru** [comment]
Hi! I don't see anything wrong with any of your macros, but to investigate, I would need to know a few things about your setup. All but one of them were explicitly listed in the issue template which you deleted. The last thing is a URL of a feed I could use for testing.

The template can be found [here](https://github.com/newsboat/newsboat/blob/master/.github/ISSUE_TEMPLATE). If you update the issue within a week, we can talk, otherwise I'll just close this.

**Comment by tamis-laan** [comment]
**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**:
```
newsboat 2.19.0
System: Linux 5.5.18-3-MANJARO (x86_64)
Compiler: g++ 9.3.0
ncurses: ncurses 6.2.20200212 (compiled with 6.2)
libcurl: libcurl/7.69.1 OpenSSL/1.1.1f zlib/1.2.11 libidn2/2.3.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh2/1.9.0 nghttp2/1.40.0 (compiled with 7.69.1)
SQLite: 3.31.1 (compiled with 3.31.1)
libxml2: compiled with 2.9.10
```

Config file (copy from ~/.newsboat/config or ~/.config/newsboat/config):

```
Iauto-reload yes

external-url-viewer "urlview"

bind-key j down
bind-key k up
bind-key j next articlelist
bind-key k prev articlelist
bind-key J next-feed articlelist
bind-key K prev-feed articlelist
bind-key G end
bind-key g home
bind-key d pagedown
bind-key u pageup
bind-key n next-unread
bind-key N prev-unread
bind-key D pb-download
bind-key U show-urls
bind-key x pb-delete
bind-key "SPACE" open

color listnormal color214 default
color listnormal_unread color175 default 

color listfocus color239 color223 standout bold
color listfocus_unread color223 color239 bold

color info color235 color250 
color article cyan default 

browser linkhandler
macro , open-in-browser
macro a set browser "nohup mpv --no-video %u &>/dev/null &" ; open-in-browser ; set  browser linkhandler
macro v set browser "nohup mpv -ytdl-format='bestvideo[height<=?720]+bestaudio/best'  %u &>/dev/null &" ; open-in-browser ; set  browser linkhandler
macro b set browser "qutebrowser %u &>/dev/null &" ; open-in-browser ; set browser linkhandler 
```

Steps to reproduce the issue:
1. Use config
2. Press , followed by v
3. Observe video is not opened in mpv.

**Comment by dennisschagt** [comment]
I can reproduce this issue. This is a bug in Newsboat.

Newsboat apparently [applies some formatting](https://github.com/newsboat/newsboat/blob/3b3649023fbb93b0783d486a1df14c43a881c5ed/src/view.cpp#L351-L358) to the `browser` variable when it contains `%u` which causes `<` to be replaced by `<>`.

This causes
`-ytdl-format='bestvideo[height<=?720]+bestaudio/best'`
to become
`-ytdl-format='bestvideo[height<>=?720]+bestaudio/best'`
which is definitely not correct behaviour.

Caused by me in https://github.com/newsboat/newsboat/commit/ad4708d264bf05196ececfcba86e122b8aa84b8b (PR #835).

**Comment by Minoru** [comment]
@tamis-laan, thanks for the updated report. This was indeed a bug in Newsboat; it was introduced in 2.19. The fix is in the `master` branch now, and will be released with 2.20 at the end of next month.

If you can't build from source, you can apply a workaround: put the command into a script, and use the script as your `browser`. Just don't use any redirections in the `browser` setting, that's what trips Newsboat up.
